e2e-tests/development-runtime/gatsby-browser.js

```diff
@@ -11,16 +12,17 @@ const addLogEntry = (action, location) => {
   })
 }
 
-exports.onPreRouteUpdate = ({ location }) => {
+export const onPreRouteUpdate = ({ location }) => {
   addLogEntry(`onPreRouteUpdate`, location)
 }
 
-exports.onRouteUpdate = ({ location }) => {
+export const onRouteUpdate = ({ location }) => {
   addLogEntry(`onRouteUpdate`, location)
 }
-
-exports.onPrefetchPathname = ({ pathname }) => {
+export const onPrefetchPathname = ({ pathname }) => {
   addLogEntry(`onPrefetchPathname`, pathname)
 }
 
-exports.wrapRootElement = Wrapper
+export const wrapRootElement = ({ element }) => (
+  <WrapRootElement element={element} />
+)
```

Comment:
This is actually important - whatever we pass to `wrapRootElement` (or `wrapPageElement`) is treated as component. We just straight call this API (and not use `React.createElement`) - that's because this is gatsby-browser API hook and we also pass pluginOptions as second parameter. Because of above this is actually not represented as React Component in react tree and react-refresh won't actually apply HMR to it (even tho webpack plugin would treat Wrapper as react component and register it as such). On top of problems there are multiple exports in `gatsby-browser` and that is also no-no for react-refresh.

This might need some additional notes in https://www.gatsbyjs.com/docs/reference/config-files/gatsby-browser/#wrapPageElement (and maybe migration guide?)